lib/ddtrace/span.rb

```diff
@@ -80,7 +83,19 @@ def initialize(tracer, name, options = {})
     #
     #   span.set_tag('http.method', request.method)
     def set_tag(key, value = nil)
-      @meta[key] = value.to_s
+      # Keys must be unique between tags and metrics
+      @metrics.delete(key) if @metrics.key?(key)
+
+      # NOTE: Adding numeric tags as metrics is stop-gap support
+      #       for numeric typed tags. Eventually they will become
+      #       tags again.
+      # Any numeric that is not an integer greater than max size is logged as a metric.
+      # Everything else gets logged as a tag.
+      if value.is_a?(Numeric) && !(value.is_a?(Integer) && value > MAX_NUMERIC_TAG_SIZE)
```

Comment:
From @lraucy's comment:
> numbers greater than 2^53 (in absolute value) must be put in Meta

we also have to handle negative values smaller than MAX_NUMERIC_TAG_SIZE.